--- a/src/NetworkLogger.cpp
+++ b/src/NetworkLogger.cpp
@@ -633,6 +633,6 @@ PyObject* PyEnableNetworkLogging( PyObject* self, PyObject* args )
 		bool ret = CCP::SetupNetworkLogging(  serverName, serverPort, (CCP::LogType) threshold, role, errorMessage, errorMessageBufferSize );
 		if (!ret)
 		{
-			return PyErr_SetString(PyExc_RuntimeError, errorMessage), 0;
+			return PyErr_SetString(PyExc_RuntimeError, errorMessage), nullptr;
 		}
 		Py_RETURN_TRUE;
